cmake/compiler/gcc/target_freestanding.cmake

```diff
@@ -0,0 +1,8 @@
+
+macro(toolchain_cc_freestanding_options)
+
+  if("${ARCH}" STREQUAL "posix")
```

Comment:
This if-statement should be dropped, it is up to the user to determine for which ARCH's that freestanding should be turned off. This is already handled because arch/posix/CMakeLists.txt is only executed when ARCH STREQUAL "posix".